core/translate/group_by.rs

```diff
@@ -146,25 +157,60 @@ pub fn emit_group_by<'a>(
         sort_cursor,
         reg_group_exprs_cmp,
         reg_subrtn_acc_clear_return_offset,
-        reg_group_exprs_acc,
+        reg_non_aggregate_exprs_acc,
         reg_abort_flag,
         reg_sorter_key,
         label_subrtn_acc_clear,
         label_acc_indicator_set_flag_true,
+        non_group_by_non_agg_column_count,
         ..
     } = *t_ctx.meta_group_by.as_mut().unwrap();
-
     let group_by = plan.group_by.as_ref().unwrap();
 
-    // all group by columns and all arguments of agg functions are in the sorter.
-    // the sort keys are the group by columns (the aggregation within groups is done based on how long the sort keys remain the same)
-    let sorter_column_count = group_by.exprs.len()
-        + plan
-            .aggregates
-            .iter()
-            .map(|agg| agg.args.len())
-            .sum::<usize>();
-    // sorter column names do not matter
+    let agg_args_count = plan
+        .aggregates
+        .iter()
+        .map(|agg| agg.args.len())
+        .sum::<usize>();
+    let group_by_count = group_by.exprs.len();
+    let non_group_by_non_agg_column_count = non_group_by_non_agg_column_count.unwrap();
+
+    // We have to know which group by expr present in resulting set
+    let group_by_expr_in_res_cols: Vec<bool> = group_by
+        .exprs
+        .iter()
+        .map(|expr| {
+            plan.result_columns
+                .iter()
+                .any(|e| exprs_are_equivalent(&e.expr, expr))
+        })
+        .collect();
```

Comment:
This section would probably be faster if we just computed `is_in_result` ad hoc:

```rust
    // Map GROUP BY columns that are in the result set to registers
    for i in 0..group_by.exprs.len() {
        let is_in_result = plan.result_columns.iter()... // etc
        if is_in_result {
            column_register_mapping.insert(i, next_reg);
            next_reg += 1;
        }
    }
    
    ...
    
    for (i, expr) in group_by.exprs.iter().enumerate() {
        let is_in_result = plan... // etc
        if is_in_result {
            if let Some(reg) = &column_register_mapping.get(&i) {
                t_ctx.resolver.expr_to_reg_cache.push((expr, **reg));
            }
        }
    }
```